### PR TITLE
#760: Migrate from WebMock to VCR

### LINKED ISSUE #760: we don't use proper GitHub fake server
Currently, we stub HTTP requests to GitHub with the help of WebMock. This makes our tests pretty long and ugly. Instead, we can use a fake implementation of GitHub API as a server. Let's try to use this one: https://github.com/h1alexbel/fakehub We can try something else if we find similar solutions.

### PR DESCRIPTION
**Pull request by VasilevNStas**
Replaced WebMock HTTP stubs with VCR cassettes across all 28 test files. 172 cassettes capture GitHub API interactions for deterministic replay.

### Key changes
- Remove ~3910 lines of WebMock stubs
- Add VCR config with `allow_playback_repeats: true` (needed for `unmask_repos`)
- Manually created 4 cassettes for edge cases (dynamic SHA, loops, sequential responses)
- Add `rate_limit_up` to tests that lacked it

### Scope
Test infrastructure migration only. No production code changes:
- `github-events.rb` refactoring reverted — extracted to a separate PR
- Migration scripts (`_*.rb`) and `MIGRATION.md` removed from repo root

Closes #760

### CODE REVIEW

**Comment by VasilevNStas** [comment]
"All 13 checks are passing now. This is my first contribution to the project — migrated tests from WebMock to VCR per #760. Ready for review!"

"Сейчас проходят все 13 проверок. Это мой первый вклад в проект — перенесенные тесты из WebMock в VCR на #760. Готов к просмотру!"

File: _migrate.rb [hunk 1]
```diff
@@ -0,0 +1,621 @@
+# frozen_string_literal: true
```
Comment (edmoffo) [inline _migrate.rb]: The five `_*.rb` scripts at the repo root (`_migrate.rb` 621 lines, `_gen_all.rb` 313, `_rewrite.rb` 204, `_fix_multiline.rb` 143, `_gen_cassettes.rb` 66; 1347 lines total) are one-off migration tools. They have no `require` callers inside `judges/`, `lib/`, or `test/`, are not invoked by any Rake task or workflow, and the leading underscore signals throwaway code. Committing them as tracked files at the repo root means they ship inside `yegor256/judges-action:latest` and stay in every checkout forever. Delete them before merge, or move them under a `tools/` directory that is excluded from the Docker image build context and from the gem file list.

File: judges/github-events/github-events.rb [hunk 2]
```diff
@@ -174,22 +174,22 @@ def self.fill(fact, json)
...
               "(transient, will retry next cycle): #{e.class}: #{e.message}"
             )
-            nil
+            []
           end
-        fact.review = review[:submitted_at] if review
+        fact.review = reviews.first[:submitted_at] if reviews.any?
         author = pl.dig(:user, :id)
-        fact.suggestions = Jp.count_suggestions(rname, fact.issue, author) if author
+        fact.suggestions = Jp.count_suggestions(rname, fact.issue, author, reviews) if author
```
Comment (edmoffo) [inline judges/github-events/github-events.rb]: This hunk is a production refactor, not a test change: it hoists `pull_request_reviews` out of the `review` block, switches the return shape from `.first` to the full array, and threads the array into `count_suggestions`. The PR title and body promise a WebMock-to-VCR migration. A behavioral change that alters the API call pattern and the data passed to `count_suggestions` should travel in its own pull request, with its own issue, so the review can focus on the call-shape change and on whether the cassette in `test/vcr_cassettes/github-events/` exercises the new path. Revert this file here and open a follow-up PR.

File: MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
+<!--
```
Comment (edmoffo) [inline MIGRATION.md]: The 184-line `MIGRATION.md` repeats material that already belongs in the PR description: the before/after snippet, the FakeHub-vs-VCR rationale, and the cassette layout. A long-form note about one migration locks one-shot context into every future checkout of the repo. Move the contents into the PR body and drop the file; if a permanent pointer is needed, a one-line note in `test/README.md` referencing the merged PR number is enough.

**Review by edmoffo — CHANGES_REQUESTED** [review]
The cassette migration itself looks careful, but the diff ships three things that should not land together. Three inline comments flag the migration scripts left at the repo root, the production refactor in `github-events.rb` smuggled into a test-tooling PR, and the `MIGRATION.md` note that duplicates the PR body. Splitting the production hunk into its own pull request and dropping the root-level scripts before merge would let this change be judged on the cassette migration alone.

**Comment by yegor256** [comment]
@VasilevNStas please, address comments above and resolve merge conflicts

**Comment by VasilevNStas** [comment]
All three review comments have been addressed in commit daa5595:

1. **Migration scripts (`_*.rb`)** — deleted from repo root
2. **Production refactor in `github-events.rb`** — reverted to master, extracted into a separate PR
3. **`MIGRATION.md`** — removed

Merge conflicts with `upstream/master` are also resolved. Could you please re-review?

**Comment by VasilevNStas** [comment]
@edmoffo All three concerns from your review have already been addressed on this branch:

1. **Root-level migration scripts** (_*.rb) — deleted
2. **github-events.rb production refactor** — reverted, no changes from master
3. **MIGRATION.md** — removed

The diff is now scoped to cassette migration + necessary config changes. Please re-review when you have a moment.

**Comment by VasilevNStas** [comment]
@edmoffo All three concerns from your review have been addressed in commit daa5595 — migration scripts deleted, github-events.rb reverted to master, MIGRATION.md removed. The diff is now scoped to cassette migration only. Could you please re-review when you have a moment?

**Comment by VasilevNStas** [comment]
@yegor256 @edmoffo please re-review

**Comment by yegor256** [comment]
@VasilevNStas merge conflicts here

**Comment by VasilevNStas** [comment]
@yegor256 plz re-review

**Comment by yegor256** [comment]
@VasilevNStas conflicts here

**Comment by VasilevNStas** [comment]
Renamed branch to `760-vcr-tests` (naming convention — should start with issue number). Replaced by new PR.

**Comment by VasilevNStas** [comment]
Replaced by #1741 (branch renamed from `fix/760-vcr-tests` → `760-vcr-tests` to follow naming convention — branch names should start with the issue number).
